modules/storages/spec/services/storages/file_links/copy_file_links_service_spec.rb

```diff
@@ -0,0 +1,88 @@
+#-- copyright
+#++
+
+require "spec_helper"
+require_module_spec_helper
+
+RSpec.describe Storages::FileLinks::CopyFileLinksService, :webmock do
+  let(:source) { create(:nextcloud_storage_with_complete_configuration) }
+  let(:target) { create(:nextcloud_storage_with_complete_configuration) }
+
+  let(:source_storage) { create(:project_storage, storage: source, project_folder_mode: "automatic") }
+  let(:target_storage) { create(:project_storage, storage: target, project_folder_mode: "automatic") }
+
+  let(:source_wp) { create_list(:work_package, 5, project: source_storage.project) }
+  let(:target_wp) { create_list(:work_package, 5, project: target_storage.project) }
+
+  let(:source_links) { source_wp.map { create(:file_link, container: _1, storage: source) } }
+
+  let(:wp_map) { source_wp.map(&:id).zip(target_wp.map(&:id)).to_h }
+
+  let(:user) { create(:user) }
+
+  subject(:service) { described_class.new(source: source_storage, target: target_storage, user:, work_packages_map: wp_map) }
+
+  before { source_links }
+
+  context "when unmanaged storage" do
+    let(:source_storage) { create(:project_storage, storage: source, project_folder_mode: "manual") }
+    let(:target_storage) { create(:project_storage, storage: target, project_folder_mode: "manual") }
+
+    it "creates file links pointing to the same files" do
+      expect { service.call }.to change(Storages::FileLink, :count).by(5)
+
+      Storages::FileLink.last(5).each_with_index do |link, index|
+        expect(link.origin_id).to eq(source_links[index].origin_id)
+      end
+    end
+  end
+
+  context "when AMPF is enabled" do
+    let(:files_info) { class_double(Storages::Peripherals::StorageInteraction::Nextcloud::FilesInfoQuery) }
+    let(:file_path_to_id) { class_double(Storages::Peripherals::StorageInteraction::Nextcloud::FilePathToIdMapQuery) }
+    let(:auth_strategy) do
+      Storages::Peripherals::StorageInteraction::AuthenticationStrategies::Strategy.new(key: :basic_auth)
```

Comment:
🟡 why not the "usual" usage as `Storages::Peripherals::StorageInteraction::AuthenticationStrategies::BasicAuth.strategy`?

Reply:
I want the actual strategy for matching the expectations. Using the `.strategy` will create a new object every time and won't match later on.